### ISSUE TITLE
Default group for "normal" user

### ISSUE THREAD

**Issue by timokau** [NixOS/nixpkgs#30427]
## Issue description

When creating a new user like this:

```
users.extraUsers.myUser = {
  isNormalUser = true;
};
```

the user `myUser` will have the generic group `users`.

This deviates from the default of most distros and the default of `useradd`:

```
-g, --gidGROUP
   The group name or number of the user's initial login group. The group name must
   exist. A group number must refer to an already existing group.

   If not specified, the behavior of useradd will depend on the USERGROUPS_ENAB
   variable in /etc/login.defs. If this variable is set to yes (or -U/--user-group
   is specified on the command line), a group will be created for the user, with
   the same name as her loginname. If the variable is set to no (or
   -N/--no-user-group is specified on the command line), useradd will set the
   primary group of the new user to the value specified by the GROUP variable in
   /etc/default/useradd, or 100 by default.
```

which can be implemented in nix like this:

```
users.extraGroups.myUser.gid = 1000;
users.extraUsers.myUser = {
  isNormalUser = true;
  uid = 1000;
};
```

Why does nixos handle this differently? I found this surprising.

One of the consequences of this is that by default all users can read everybodys
files.

### Steps to reproduce

Create a new user, observe that his group is "users".

## Technical details

* System: 17.09.1535.1fdca25ee8 (Hummingbird)
* Nix version: 1.11.15
* Nixpkgs version: 17.09.1535.1fdca25ee8
* Sandboxing enabled: false

**Comment by peterhoeg** [comment]
> One of the consequences of this is that by default all users can read everybodys files.

Isn't your user home directory set to 0700 ?

**Comment by NeQuissimus** [comment]
Actually, look at this:

This is my user config
```nix
users.extraUsers.kubernetes = {
    extraGroups = [ "docker" ];
    isNormalUser = true;
    name = "kubernetes";
    uid = 1002;
  };
  users.extraUsers.nix = {
    extraGroups = [ "docker" ];
    isNormalUser = true;
    name = "nix";
    uid = 1003;
  };
  users.extraUsers.ruby = {
    extraGroups = [ "docker" ];
    isNormalUser = true;
    name = "ruby";
    uid = 1004;
  };
  users.extraUsers.scala = {
    extraGroups = [ "docker" ];
    isNormalUser = true;
    name = "scala";
    uid = 1001;
  };
  users.extraUsers.metron = {
    extraGroups = [ "docker" ];
    isNormalUser = true;
    name = "metron";
    uid = 1005;
  };
```

Now check this  (nequi is my user I mess around with, ignore that one):
All but `metron` were created on 17.03 (upgraded to 17.09).
`metron` was created on 17.09. I think something has changed in the permission set:

```bash
λ ls -la /home
drwx------ - kubernetes 25 Sep  8:50 kubernetes
drwxr-xr-x - metron     11 Oct  9:45 metron
drwxr-xr-x - nequi      18 Oct 11:11 nequi
drwx------ - nix        18 Oct  8:20 nix
drwx------ - ruby       28 Sep  7:34 ruby
drwx------ - scala      18 Oct 11:11 scala
```

**Comment by yegortimoshenko** [comment]
I don't think this can be fixed in a reverse-compatible way.

Permission bit for user directories is fixed and not an issue anymore in current `17.09`.

**Comment by peterhoeg** [comment]
Closing due to:

> Permission bit for user directories is fixed and not an issue anymore in current 17.09.
